jerry-core/parser/js/js-parser.c

```diff
@@ -3128,27 +3128,18 @@ parser_parse_script (const uint8_t *arg_list_p, /**< function argument list */
       return NULL;
     }
 
-#if ENABLED (JERRY_ERROR_MESSAGES)
-    ecma_string_t *err_str_p;
-
-#if !ENABLED (JERRY_ESNEXT)
     if (parser_error.error == PARSER_ERR_INVALID_REGEXP)
     {
-      ecma_value_t error = jcontext_take_exception ();
-      ecma_property_t *prop_p = ecma_find_named_property (ecma_get_object_from_value (error),
-                                                          ecma_get_magic_string (LIT_MAGIC_STRING_MESSAGE));
-      ecma_free_value (error);
-      JERRY_ASSERT (prop_p);
-      err_str_p = ecma_get_string_from_value (ECMA_PROPERTY_VALUE_PTR (prop_p)->value);
-      ecma_ref_ecma_string (err_str_p);
-    }
-    else
-#endif /* !ENABLED (JERRY_ESNEXT) */
-    {
-      const lit_utf8_byte_t *err_bytes_p = (const lit_utf8_byte_t *) parser_error_to_string (parser_error.error);
-      lit_utf8_size_t err_bytes_size = lit_zt_utf8_string_size (err_bytes_p);
-      err_str_p = ecma_new_ecma_string_from_utf8 (err_bytes_p, err_bytes_size);
+      /* The RegExp compiler has already raised an exception. */
+      JERRY_ASSERT (jcontext_has_pending_exception ());
```

Comment:
The parser were able to report the error thrown by the regexp engine as syntax error in ES5.1 profile which must be kept new in the ES.NEXT profile since the invalid regexp behaviors in the two standard revisions are the same.